packages/frontend/src/components/ui/AutoComplete/AutoComplete.tsx

```diff
@@ -40,22 +44,54 @@ export const AutoComplete = <T extends object>({
   const filteredData = useMemo(
     () => (query === ''
       ? data
-      : data.filter((item) =>
+      : data.filter((item) => {
+        if (optionKeys.value) {
+          // @ts-ignore
+          return item[optionKeys.value]
+            .toLowerCase()
+            .includes(query.toLowerCase());
+        }
+        return (
+          item[optionKeys.label]
+            // @ts-ignore
+            .toLowerCase()
+            .includes(query.toLowerCase())
+        );
+      })),
+    [query, data, optionKeys.value, optionKeys.label],
+  );
+
+  const changedValueHandler = useCallback(
+    (value: T) => {
+      console.log(value);
+      onSelect(value, query);
+      if (!value) {
+        return;
+      }
+      if (typeof value === 'object') {
         // @ts-ignore
-        // eslint-disable-next-line implicit-arrow-linebreak
-        item[optionKeys.value].toLowerCase().includes(query.toLowerCase()))),
-    [query, data, optionKeys.value],
+        setSelectedOption(value[optionKeys.label]);
+      } else {
+        setSelectedOption(value);
+      }
+    },
+    [onSelect, optionKeys.label, query],
   );
 
+  useEffect(() => {
+    const timeout = setTimeout(changedValueHandler, DEBOUNCE_DELAY);
+
+    return () => {
+      clearTimeout(timeout);
+    };
```

Comment:
can you use the useDebounce hook we have here?

Reply:
I've considered to use the hook, but I don't think it would work well with functions and this use case since functions are not **primitive values** so therefore it may not be ideal - wdyt?

Reply:
It's quite hard to follow the code. The useEffect will be called whenever the deps change. The deps change when `changedValueHandler` is called. `changedValueHandler` is called when then combo-box changes. So it seems like `changedValueHandler` is going to get called multiple times each time the combo box is changed.

Ideally you want to use `useDebounce` and pass in a value. `useDebounce` will give you back the value debounced. When then debounced value changes, you can use it inside a useEffect or call a callback prop with then value.

Reply:
Changes made!